src/services/unused-address.service.ts

```diff
@@ -19,30 +19,34 @@ export class UnusedAddressService {
     this.logger = getLogger('unused-address');
   }
 
-  public isUnusedAddresses(addresses: string[]): Promise<UnusedAddressResponse> {
-    this.logger.debug("[isUnusedAddresses] starting to analyse addresses");
-    const response: UnusedAddressResponse =  new UnusedAddressResponse({ data: [] });
-    const addressInfoPromises = [];
-    for (const address of addresses) {
-       addressInfoPromises.push(this.bitcoinService.getAddressInfo(address));
-    }
-    return Promise.all(addressInfoPromises)
-    .then((bitcoinAddressList: BitcoinAddress[]) => {
-      for (const bitcoinAddress of bitcoinAddressList) {
-        const totalReceived: SatoshiBig = new SatoshiBig(bitcoinAddress.totalReceived, 'satoshi');
-        const totalSent: SatoshiBig = new SatoshiBig(bitcoinAddress.totalSent, 'satoshi');
-        const unused = totalReceived.eq(0) && totalSent.eq(0);
-        response.data.push(new AddressUsedStatus({
-          address: bitcoinAddress.address,
-          unused,
-        }));
-        this.logger.trace(`[isUnusedAddresses] ${bitcoinAddress} is ${unused ? 'unused' : 'used'}`);
+  public async isUnusedAddresses(addresses: string[]): Promise<UnusedAddressResponse> {
+        this.logger.debug("[isUnusedAddresses] starting to analyse addresses");
+        const response: UnusedAddressResponse =  new UnusedAddressResponse({ data: [] });
+        const addressInfoPromises = [];
+        for (const address of addresses) {
+          try {
+           addressInfoPromises.push(await this.bitcoinService.getAddressInfo(address));
+          } catch(e) {
+            this.logger.debug(`[isUnusedAddresses] Error obtaining one address, keep trying to get everyone else. ${e}`);
+          }
+        }
+        return Promise.all(addressInfoPromises)
+        .then((bitcoinAddressList: BitcoinAddress[]) => {
```

Comment:
If you are going to await for each address then don't do a Promise.all.
Choose one or the other